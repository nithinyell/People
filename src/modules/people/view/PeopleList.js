// People List View

import {FlatList, StyleSheet, Text, TouchableOpacity, View} from 'react-native';
import {PeopleViewModel} from '../view-model/people-viewmodel';
import {Avatar, Card} from 'react-native-paper';
import {getRandomColor} from '../../utils/get-random-color';

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  person: {
    flex: 1,
    padding: 5,
    fontSize: 14,
    height: 27,
    lineHeight: 20,
  },
  email: {
    flex: 1,
    padding: 5,
    fontSize: 12,
    height: 27,
    lineHeight: 20,
  },
});

const PeopleList = ({navigation}) => {
  const { people } = PeopleViewModel();

  const navigateToDetails = person => {
    navigation.navigate('Person Details', {person});
  };

  const PersonItem = ({person}) => {
    return (
      <TouchableOpacity
        onPress={() => {
          navigateToDetails(person);
        }}>
        <View style={{padding: 5, flex: 1}}>
          <Card
            style={{
              marginTop: 0,
              flexDirection: 'row',
              alignItems: 'center',
            }}>
            <Card.Content style={{flexDirection: 'row', alignItems: 'center'}}>
              <Avatar.Text
                size={45}
                label={person.name[0].toUpperCase()}
                style={{marginRight: 8, backgroundColor: getRandomColor()}}
              />
              <View style={{flexDirection: 'column', padding: 5}}>
                <Text style={styles.person}>{person.name}</Text>
                <Text style={styles.email}>{person.email}</Text>
              </View>
            </Card.Content>
          </Card>
        </View>
      </TouchableOpacity>
    );
  };

  return (
    <View style={[styles.center]}>
      <FlatList
        data={people}
        renderItem={({item}) => <PersonItem person={item} />}
        keyExtractor={item => item.email}
      />
    </View>
  );
};

export default PeopleList;