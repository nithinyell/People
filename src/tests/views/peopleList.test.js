import PeopleList from '../../modules/people/view/PeopleList';
import {render, act, waitFor, screen} from '@testing-library/react-native';

describe('PeopleList', () => {
  test('renders list of people', async () => {
    await act(async () => {
      // Render the PeopleList component
      await waitFor(() => render(<PeopleList />));

      await waitFor(() => {
        // Check if the names of Chris is present in the rendered component
        expect(screen.getByText('Chris Kong')).toBeTruthy();
        expect(screen.getByText('christ.kong@email.com')).toBeTruthy();

        // Check if a non-existing name is not present
        expect(screen.queryByText('Non Existing Name')).toBeNull();
      });
    });
  });
});