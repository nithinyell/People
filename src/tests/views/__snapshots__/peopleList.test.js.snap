// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    [
      undefined,
    ]
  }
>
  <RCTScrollView
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
</View>
`;
